Revert to old license syntax for compatability with older setuptools

author: micketeer@gmail.com
--- pyproject.toml.orig	2025-07-02 22:22:31.480281841 +0000
+++ pyproject.toml	2025-07-02 22:22:52.293048468 +0000
@@ -10,8 +10,7 @@
 description = "Backported and Experimental Type Hints for Python 3.9+"
 readme = "README.md"
 requires-python = ">=3.9"
-license = "PSF-2.0"
-license-files = ["LICENSE"]
+license = { file = "LICENSE" }
 keywords = [
     "annotations",
     "backport",
